classes/AddressChecksumCore.php

```diff
@@ -49,7 +49,7 @@ public function generateChecksum($address)
         try {
             $fields = $address->getFields();
         } catch (Exception $e) {
-            throw new Exception($e->getMessage());
+            throw new Exception($this->trans('Your address is incomplete, please update it.', array(), 'Shop.Notifications.Error'));
```

Comment:
Yes, you shouldn't redefine the `trans()` method. We must have a unique one.
You can keep the `getTranslator()` function but can you change its visibility to `private`, please?